deployment-configuration/compose/templates/auto-gatekeepers.yaml

```diff
@@ -0,0 +1,146 @@
+{{/* Secured Services/Deployments */}}
+{{- define "securedservice.deploy" }}
+{{- $tls := not (not .root.Values.tls) }}
+{{- $gk_name := printf "%s-gk" .app.harness.service.name }}
+{{ $gk_name }}:
+  networks:
+  - ch
+  restart: always
+  image: quay.io/gogatekeeper/gatekeeper:1.3.8
+  expose:
+    - '8080'
+    - '8443'
+  deploy:
+    mode: replicated
+    replicas: 1
+    resources:
+      limits:
+        cpus: 100m
+        memory: 64M
+      reservations:
+        cpus: 50m
+        memory: 32M
+  environment:
+    - PROXY_CONFIG_FILE=/opt/proxy.yml
+  volumes:
+    - ./compose/resources/generated/{{ $gk_name }}/proxy.yml:/opt/proxy.yml
+    - ./compose/resources/generated/{{ $gk_name }}/cacert.crt:/etc/pki/ca-trust/source/anchors/cacert.crt
+    - ./compose/resources/generated/{{ $gk_name }}/access-denied.html.tmpl:/templates/access-denied.html.tmpl
+  labels:
+    - "traefik.enable=true"
+    - "traefik.http.services.{{ $gk_name }}.loadbalancer.server.port={{ .app.harness.service.port }}"
+    - "traefik.http.routers.gatekeeper.middlewares=redirect-middleware"
+    - "traefik.http.routers.{{ $gk_name }}.rule=Host(`{{ .app.harness.subdomain }}.{{ .root.Values.domain }}`)"
+    - "traefik.http.routers.{{ $gk_name }}.entrypoints=web"
+  depends_on:
+    accounts:
+      condition: service_healthy
+  links:
+    - traefik:accounts.{{ .root.Values.domain }}
+    - {{ .app.harness.service.name }}:{{ .app.harness.service.name }}.default
+{{- end }}
+
+{{- define "securedservice.deploy.resources" }}
+{{- $tls := not (not .root.Values.tls) }}
+cloudharness-metadata:
+  path: resources/generated/{{ .app.harness.service.name }}-gk/proxy.yml
+
+data: |-
+    verbose: {{ .root.Values.debug }}
```

Comment:
So much duplication here, we should find a way to use the same source template for this

Reply:
In this kind of context, and same for the other files embedded in this same file, we could have a shared template folder somewhere where the content of the files that needs to be generate for k8s and docker-compose are added, and then included inside of the respective templates (for helm and docker-compose)? 

The way of declaring the content of a file is not the same between the k8s target and docker-compose target, so the only thing that we can "share" there is the content of the file itself, not the way it will be generated (I'm not sure I'm very clear in what I'm saying)